src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -3981,17 +3986,137 @@ public Derived() : this(0)
             }
             """;
 
+        var comp = CreateCompilationWithRequiredMembers(new[] { derived, @base });
+
+        comp.VerifyDiagnostics(
+            // (10,12): warning CS8618: Non-nullable property 'Prop3' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+            //     public Derived(int unused) : base()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "Derived").WithArguments("property", "Prop3").WithLocation(10, 12),
+            // (10,12): warning CS8618: Non-nullable property 'Prop1' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+            //     public Derived(int unused) : base()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "Derived").WithArguments("property", "Prop1").WithLocation(10, 12),
+            // (15,24): error CS9039: This constructor must add 'SetsRequiredMembers' because it chains to a constructor that has that attribute.
+            //     public Derived() : this(0)
+            Diagnostic(ErrorCode.ERR_ChainingToSetsRequiredMembersRequiresSetsRequiredMembers, "this").WithLocation(15, 24)
+        );
+
+        var baseComp = CreateCompilationWithRequiredMembers(@base);
+        comp = CreateCompilation(derived, new[] { baseComp.EmitToImageReference() });
+        comp.VerifyDiagnostics(
```

Comment:
got it now based on offline conversation. When the base property is from metadata we don't know about the associated field and so don't give the property a maybe-null initial state. Therefore there is no warning on exit. Sounds like we will discuss whether to change that initial state separately from this PR.

Reply:
https://github.com/dotnet/csharplang/issues/6359 is the issue for discussion in LDM tomorrow.